modules/dlm/src/main/java/org/elasticsearch/dlm/action/TransportGetDataLifecycleAction.java

```diff
@@ -0,0 +1,101 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.dlm.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeReadAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.DataLifecycle;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.datastreams.action.DataStreamsActionUtil;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Collects the data streams from the cluster state, filters the ones that do not have a lifecycle configured and then returns
+ * a list of the data stream name and respective lifecycle configuration.
+ */
+public class TransportGetDataLifecycleAction extends TransportMasterNodeReadAction<
+    GetDataLifecycleAction.Request,
+    GetDataLifecycleAction.Response> {
+    private final ClusterSettings clusterSettings;
+
+    @Inject
+    public TransportGetDataLifecycleAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            GetDataLifecycleAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            GetDataLifecycleAction.Request::new,
+            indexNameExpressionResolver,
+            GetDataLifecycleAction.Response::new,
+            ThreadPool.Names.SAME
+        );
+        clusterSettings = clusterService.getClusterSettings();
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        GetDataLifecycleAction.Request request,
+        ClusterState state,
+        ActionListener<GetDataLifecycleAction.Response> listener
+    ) {
+        List<String> results = DataStreamsActionUtil.getDataStreamNames(
+            indexNameExpressionResolver,
+            state,
+            request.getNames(),
+            request.indicesOptions()
+        );
+        Map<String, DataStream> dataStreams = state.metadata().dataStreams();
+
+        listener.onResponse(
+            new GetDataLifecycleAction.Response(
+                results.stream()
+                    .map(dataStreams::get)
```

Comment:
I think we need a filter nullables here? In case a data stream that doesn't exist was requested ?

Reply:
I can add it as a fail-safe in case something goes wrong. I think it's redundant because the `DataStreamsActionUtil::getDataStreamNames` and the `state.metadata().dataStreams()` use the same `state` so I would assume that there is no chance that a data stream will exist in one and not the other case. Do you agree with this assumption? And if yes, do you still think we should add `.filter(Objects::nonNull)`?

Reply:
++ you're right it would be redundant. I think it could be valuable as the assumption that `getDataStreamNames` is always called before this piece of code is implicit.